battle/boss.py

```diff
@@ -799,24 +873,24 @@ def spherimorph():
                     # Except we haven't learned Steal yet. That's no good.
                     #    _steal()
                     # else:
-                    battle.main.defend()
+                    CurrentPlayer().defend()
                     kim_turn = True
-                elif 6 not in memory.main.get_active_battle_formation():
+                elif not Rikku.active():
                     battle.main.buddy_swap(Rikku)
-                elif 5 not in memory.main.get_active_battle_formation():
+                elif not Lulu.active():
                     battle.main.buddy_swap(Lulu)
                 else:
-                    battle.main.defend()
+                    CurrentPlayer().defend()
             elif Lulu.is_turn():
                 if not battle.main.spheri_spell_item_ready():
                     if spell_num == 1:
-                        battle.main.ice()
+                        CurrentPlayer().cast_black_magic_spell(3)
```

Comment:
This has gotten less easy to read. Lol. In a follow up PR, it would likely be worth converted `3` to an Enum.

Reply:
Agreed, and I stated this in the PR description :P

Reply:
A lot of the CurrentPlayers() can be updated to be the actual character, but this was the easiest way to do it w/ doing a mass conversion.